[libdatachannel] Link failure in global mode

### Operating system

Windows 11

### Compiler

MSVC (VS 2022)

### Steps to reproduce the behavior

```Shell
- ./vcpkg install libdatachannel:x86-windows
 - Write any piece of code using libdatachannel
 - Receive LNK2019
```


### Failure logs

```
1>main.obj : error LNK2019: ссылка на неразрешенный внешний символ "__declspec(dllimport) public: __thiscall rtc::IceServer::IceServer(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &)" (__imp_??0IceServer@rtc@@QAE@ABV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@@Z) в функции "public: static void __cdecl std::_Default_allocator_traits<class std::allocator<struct rtc::IceServer> >::construct<struct rtc::IceServer,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > &>(class std::allocator<struct rtc::IceServer> &,struct rtc::IceServer * const,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > &)" (??$construct@UIceServer@rtc@@AAV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@@?$_Default_allocator_traits@V?$allocator@UIceServer@rtc@@@std@@@std@@SAXAAV?$allocator@UIceServer@rtc@@@1@QAUIceServer@rtc@@AAV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@1@@Z).

1>main.obj : error LNK2019: ссылка на неразрешенный внешний символ "__declspec(dllimport) public: __thiscall rtc::IceServer::~IceServer(void)" (__imp_??1IceServer@rtc@@QAE@XZ) в функции "public: void * __thiscall rtc::IceServer::`scalar deleting destructor'(unsigned int)" (??_GIceServer@rtc@@QAEPAXI@Z).
```

### Additional context

I'm sorry in advance for logs in russian, but I hope the error codes will be enough to grasp what happens. Also I don't know if I filled the issue blank correctly, so feel free to correct me. So here is how it went

I tried recreating the [example from libdatachannels github](https://github.com/paullouisageneau/libdatachannel/blob/master/examples/streamer) with x86-windows triplet, but received linkage errors as shown in the Failure logs.

Then yesterday I saw, that someone had a similar issue which was corrected in [#40337](https://github.com/microsoft/vcpkg/pull/40337) , so i removed vcpkg completely as vcpkg upgrade is not advised to use, and reinstalled it. All the other dependencies in my main project installed via vcpkg work perfectly fine, but the issue with libdatachannel persists. 

I tried also using the static triplet `x86-windows-static`, set `vcpkg use static libraries` to `Yes` and changed `/MDd` to `/MTd`, and got a bit different kind of errors
```1>main.obj : error LNK2019: ссылка на неразрешенный внешний символ "public: __thiscall rtc::RtpPacketizationConfig::RtpPacketizationConfig(unsigned int,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> >,unsigned char,unsigned int,unsigned char)" (??0RtpPacketizationConfig@rtc@@QAE@IV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@EIE@Z) в функции "void __cdecl std::_Construct_in_place<class rtc::RtpPacketizationConfig,unsigned int const &,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,unsigned char const &,unsigned int const &>(class rtc::RtpPacketizationConfig &,unsigned int const &,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,unsigned char const &,unsigned int const &)" (??$_Construct_in_place@VRtpPacketizationConfig@rtc@@ABIABV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@ABEABI@std@@YAXAAVRtpPacketizationConfig@rtc@@ABIABV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@0@ABE1@Z).
1>main.obj : error LNK2019: ссылка на неразрешенный внешний символ "public: double __thiscall rtc::RtpPacketizationConfig::timestampToSeconds(unsigned int)" (?timestampToSeconds@RtpPacketizationConfig@rtc@@QAENI@Z) в функции "public: __thiscall `class std::shared_ptr<class Stream> __cdecl createStream(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> >,unsigned int,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> >)'::`2'::<lambda_1>::operator()(enum Stream::StreamSourceType,unsigned __int64,class std::vector<enum std::byte,class std::allocator<enum std::byte> >)const " (??R<lambda_1>@?1??createStream@@YA?AV?$shared_ptr@VStream@@@std@@V?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@3@I0@Z@QBE@W4StreamSourceType@Stream@@_KV?$vector@W4byte@std@@V?$allocator@W4byte@std@@@2@@3@@Z).
1>main.obj : error LNK2019: ссылка на неразрешенный внешний символ "public: unsigned int __thiscall rtc::RtpPacketizationConfig::secondsToTimestamp(double)" (?secondsToTimestamp@RtpPacketizationConfig@rtc@@QAEIN@Z) в функции "public: __thiscall `class std::shared_ptr<class Stream> __cdecl createStream(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> >,unsigned int,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> >)'::`2'::<lambda_1>::operator()(enum Stream::StreamSourceType,unsigned __int64,class std::vector<enum std::byte,class std::allocator<enum std::byte> >)const " (??R<lambda_1>@?1??createStream@@YA?AV?$shared_ptr@VStream@@@std@@V?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@3@I0@Z@QBE@W4StreamSourceType@Stream@@_KV?$vector@W4byte@std@@V?$allocator@W4byte@std@@@2@@3@@Z).
1>main.obj : error LNK2019: ссылка на неразрешенный внешний символ "public: __thiscall rtc::RtpPacketizer::RtpPacketizer(class std::shared_ptr<class rtc::RtpPacketizationConfig>)" (??0RtpPacketizer@rtc@@QAE@V?$shared_ptr@VRtpPacketizationConfig@rtc@@@std@@@Z) в функции "public: __thiscall rtc::AudioRtpPacketizer<48000>::AudioRtpPacketizer<48000>(class std::shared_ptr<class rtc::RtpPacketizationConfig>)" (??0?$AudioRtpPacketizer@$0LLIA@@rtc@@QAE@V?$shared_ptr@VRtpPacketizationConfig@rtc@@@std@@@Z).
1>main.obj : error LNK2019: ссылка на неразрешенный внешний символ "public: virtual __thiscall rtc::RtpPacketizer::~RtpPacketizer(void)" (??1RtpPacketizer@rtc@@UAE@XZ) в функции "public: virtual __thiscall rtc::AudioRtpPacketizer<48000>::~AudioRtpPacketizer<48000>(void)" (??1?$AudioRtpPacketizer@$0LLIA@@rtc@@UAE@XZ).
1>main.obj : error LNK2001: неразрешенный внешний символ "public: virtual void __thiscall rtc::RtpPacketizer::media(class rtc::Description::Media const &)" (?media@RtpPacketizer@rtc@@UAEXABVMedia@Description@2@@Z).
1>main.obj : error LNK2001: неразрешенный внешний символ "public: virtual void __thiscall rtc::RtpPacketizer::outgoing(class std::vector<class std::shared_ptr<struct rtc::Message>,class std::allocator<class std::shared_ptr<struct rtc::Message> > > &,class std::function<void __cdecl(class std::shared_ptr<struct rtc::Message>)> const &)" (?outgoing@RtpPacketizer@rtc@@UAEXAAV?$vector@V?$shared_ptr@UMessage@rtc@@@std@@V?$allocator@V?$shared_ptr@UMessage@rtc@@@std@@@2@@std@@ABV?$function@$$A6AXV?$shared_ptr@UMessage@rtc@@@std@@@Z@4@@Z).
1>main.obj : error LNK2001: неразрешенный внешний символ "protected: virtual class std::shared_ptr<struct rtc::Message> __thiscall rtc::RtpPacketizer::packetize(class std::shared_ptr<class std::vector<enum std::byte,class std::allocator<enum std::byte> > >,bool)" (?packetize@RtpPacketizer@rtc@@MAE?AV?$shared_ptr@UMessage@rtc@@@std@@V?$shared_ptr@V?$vector@W4byte@std@@V?$allocator@W4byte@std@@@2@@std@@@4@_N@Z).
1>main.obj : error LNK2019: ссылка на неразрешенный внешний символ "public: __thiscall rtc::H264RtpPacketizer::H264RtpPacketizer(enum rtc::NalUnit::Separator,class std::shared_ptr<class rtc::RtpPacketizationConfig>,unsigned short)" (??0H264RtpPacketizer@rtc@@QAE@W4Separator@NalUnit@1@V?$shared_ptr@VRtpPacketizationConfig@rtc@@@std@@G@Z) в функции "void __cdecl std::_Construct_in_place<class rtc::H264RtpPacketizer,enum rtc::NalUnit::Separator,class std::shared_ptr<class rtc::RtpPacketizationConfig> &>(class rtc::H264RtpPacketizer &,enum rtc::NalUnit::Separator &&,class std::shared_ptr<class rtc::RtpPacketizationConfig> &)" (??$_Construct_in_place@VH264RtpPacketizer@rtc@@W4Separator@NalUnit@2@AAV?$shared_ptr@VRtpPacketizationConfig@rtc@@@std@@@std@@YAXAAVH264RtpPacketizer@rtc@@$$QAW4Separator@NalUnit@2@AAV?$shared_ptr@VRtpPacketizationConfig@rtc@@@0@@Z).
1>main.obj : error LNK2019: ссылка на неразрешенный внешний символ "public: __thiscall rtc::RtcpNackResponder::RtcpNackResponder(unsigned int)" (??0RtcpNackResponder@rtc@@QAE@I@Z) в функции "void __cdecl std::_Construct_in_place<class rtc::RtcpNackResponder>(class rtc::RtcpNackResponder &)" (??$_Construct_in_place@VRtcpNackResponder@rtc@@$$V@std@@YAXAAVRtcpNackResponder@rtc@@@Z).
1>main.obj : error LNK2019: ссылка на неразрешенный внешний символ "public: __thiscall rtc::RtcpSrReporter::RtcpSrReporter(class std::shared_ptr<class rtc::RtpPacketizationConfig>)" (??0RtcpSrReporter@rtc@@QAE@V?$shared_ptr@VRtpPacketizationConfig@rtc@@@std@@@Z) в функции "void __cdecl std::_Construct_in_place<class rtc::RtcpSrReporter,class std::shared_ptr<class rtc::RtpPacketizationConfig> &>(class rtc::RtcpSrReporter &,class std::shared_ptr<class rtc::RtpPacketizationConfig> &)" (??$_Construct_in_place@VRtcpSrReporter@rtc@@AAV?$shared_ptr@VRtpPacketizationConfig@rtc@@@std@@@std@@YAXAAVRtcpSrReporter@rtc@@AAV?$shared_ptr@VRtpPacketizationConfig@rtc@@@0@@Z).
1>main.obj : error LNK2019: ссылка на неразрешенный внешний символ "public: unsigned int __thiscall rtc::RtcpSrReporter::lastReportedTimestamp(void)const " (?lastReportedTimestamp@RtcpSrReporter@rtc@@QBEIXZ) в функции "public: __thiscall `class std::shared_ptr<class Stream> __cdecl createStream(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> >,unsigned int,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> >)'::`2'::<lambda_1>::operator()(enum Stream::StreamSourceType,unsigned __int64,class std::vector<enum std::byte,class std::allocator<enum std::byte> >)const " (??R<lambda_1>@?1??createStream@@YA?AV?$shared_ptr@VStream@@@std@@V?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@3@I0@Z@QBE@W4StreamSourceType@Stream@@_KV?$vector@W4byte@std@@V?$allocator@W4byte@std@@@2@@3@@Z).
1>main.obj : error LNK2019: ссылка на неразрешенный внешний символ "public: void __thiscall rtc::RtcpSrReporter::setNeedsToReport(void)" (?setNeedsToReport@RtcpSrReporter@rtc@@QAEXXZ) в функции "public: __thiscall `class std::shared_ptr<class Stream> __cdecl createStream(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> >,unsigned int,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> >)'::`2'::<lambda_1>::operator()(enum Stream::StreamSourceType,unsigned __int64,class std::vector<enum std::byte,class std::allocator<enum std::byte> >)const " (??R<lambda_1>@?1??createStream@@YA?AV?$shared_ptr@VStream@@@std@@V?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@3@I0@Z@QBE@W4StreamSourceType@Stream@@_KV?$vector@W4byte@std@@V?$allocator@W4byte@std@@@2@@3@@Z).
1>datachannel-static.lib(icetransport.cpp.obj) : error LNK2019: ссылка на неразрешенный внешний символ __imp__juice_create в функции "public: __thiscall rtc::impl::IceTransport::IceTransport(struct rtc::Configuration const &,class std::function<void __cdecl(class rtc::Candidate const &)>,class std::function<void __cdecl(enum rtc::impl::Transport::State)>,class std::function<void __cdecl(enum rtc::impl::IceTransport::GatheringState)>)" (??0IceTransport@impl@rtc@@QAE@ABUConfiguration@2@V?$function@$$A6AXABVCandidate@rtc@@@Z@std@@V?$function@$$A6AXW4State@Transport@impl@rtc@@@Z@5@V?$function@$$A6AXW4GatheringState@IceTransport@impl@rtc@@@Z@5@@Z).
1>datachannel-static.lib(icetransport.cpp.obj) : error LNK2019: ссылка на неразрешенный внешний символ __imp__juice_destroy в функции "public: __thiscall rtc::impl::IceTransport::IceTransport(struct rtc::Configuration const &,class std::function<void __cdecl(class rtc::Candidate const &)>,class std::function<void __cdecl(enum rtc::impl::Transport::State)>,class std::function<void __cdecl(enum rtc::impl::IceTransport::GatheringState)>)" (??0IceTransport@impl@rtc@@QAE@ABUConfiguration@2@V?$function@$$A6AXABVCandidate@rtc@@@Z@std@@V?$function@$$A6AXW4State@Transport@impl@rtc@@@Z@5@V?$function@$$A6AXW4GatheringState@IceTransport@impl@rtc@@@Z@5@@Z).
1>datachannel-static.lib(icetransport.cpp.obj) : error LNK2019: ссылка на неразрешенный внешний символ __imp__juice_gather_candidates в функции "public: void __thiscall rtc::impl::IceTransport::gatherLocalCandidates(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> >,class std::vector<struct rtc::IceServer,class std::allocator<struct rtc::IceServer> >)" (?gatherLocalCandidates@IceTransport@impl@rtc@@QAEXV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@V?$vector@UIceServer@rtc@@V?$allocator@UIceServer@rtc@@@std@@@5@@Z).
1>datachannel-static.lib(icetransport.cpp.obj) : error LNK2019: ссылка на неразрешенный внешний символ __imp__juice_get_local_description в функции "public: class rtc::Description __thiscall rtc::impl::IceTransport::getLocalDescription(enum rtc::Description::Type)const " (?getLocalDescription@IceTransport@impl@rtc@@QBE?AVDescription@3@W4Type@43@@Z).
1>datachannel-static.lib(icetransport.cpp.obj) : error LNK2019: ссылка на неразрешенный внешний символ __imp__juice_set_remote_description в функции "public: void __thiscall rtc::impl::IceTransport::setRemoteDescription(class rtc::Description const &)" (?setRemoteDescription@IceTransport@impl@rtc@@QAEXABVDescription@3@@Z).
1>datachannel-static.lib(icetransport.cpp.obj) : error LNK2019: ссылка на неразрешенный внешний символ __imp__juice_add_remote_candidate в функции "public: bool __thiscall rtc::impl::IceTransport::addRemoteCandidate(class rtc::Candidate const &)" (?addRemoteCandidate@IceTransport@impl@rtc@@QAE_NABVCandidate@3@@Z).
1>datachannel-static.lib(icetransport.cpp.obj) : error LNK2019: ссылка на неразрешенный внешний символ __imp__juice_add_turn_server в функции "private: void __thiscall rtc::impl::IceTransport::addIceServer(struct rtc::IceServer)" (?addIceServer@IceTransport@impl@rtc@@AAEXUIceServer@3@@Z).
1>datachannel-static.lib(icetransport.cpp.obj) : error LNK2019: ссылка на неразрешенный внешний символ __imp__juice_send_diffserv в функции "private: virtual bool __thiscall rtc::impl::IceTransport::outgoing(class std::shared_ptr<struct rtc::Message>)" (?outgoing@IceTransport@impl@rtc@@EAE_NV?$shared_ptr@UMessage@rtc@@@std@@@Z).
1>datachannel-static.lib(icetransport.cpp.obj) : error LNK2019: ссылка на неразрешенный внешний символ __imp__juice_get_selected_candidates в функции "public: bool __thiscall rtc::impl::IceTransport::getSelectedCandidatePair(class rtc::Candidate *,class rtc::Candidate *)" (?getSelectedCandidatePair@IceTransport@impl@rtc@@QAE_NPAVCandidate@3@0@Z).
1>datachannel-static.lib(icetransport.cpp.obj) : error LNK2019: ссылка на неразрешенный внешний символ __imp__juice_get_selected_addresses в функции "public: class std::optional<class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > > __thiscall rtc::impl::IceTransport::getLocalAddress(void)const " (?getLocalAddress@IceTransport@impl@rtc@@QBE?AV?$optional@V?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@@std@@XZ).
1>datachannel-static.lib(icetransport.cpp.obj) : error LNK2019: ссылка на неразрешенный внешний символ __imp__juice_set_log_level в функции "public: __thiscall rtc::impl::IceTransport::IceTransport(struct rtc::Configuration const &,class std::function<void __cdecl(class rtc::Candidate const &)>,class std::function<void __cdecl(enum rtc::impl::Transport::State)>,class std::function<void __cdecl(enum rtc::impl::IceTransport::GatheringState)>)" (??0IceTransport@impl@rtc@@QAE@ABUConfiguration@2@V?$function@$$A6AXABVCandidate@rtc@@@Z@std@@V?$function@$$A6AXW4State@Transport@impl@rtc@@@Z@5@V?$function@$$A6AXW4GatheringState@IceTransport@impl@rtc@@@Z@5@@Z).
1>datachannel-static.lib(icetransport.cpp.obj) : error LNK2019: ссылка на неразрешенный внешний символ __imp__juice_set_log_handler в функции "public: __thiscall rtc::impl::IceTransport::IceTransport(struct rtc::Configuration const &,class std::function<void __cdecl(class rtc::Candidate const &)>,class std::function<void __cdecl(enum rtc::impl::Transport::State)>,class std::function<void __cdecl(enum rtc::impl::IceTransport::GatheringState)>)" (??0IceTransport@impl@rtc@@QAE@ABUConfiguration@2@V?$function@$$A6AXABVCandidate@rtc@@@Z@std@@V?$function@$$A6AXW4State@Transport@impl@rtc@@@Z@5@V?$function@$$A6AXW4GatheringState@IceTransport@impl@rtc@@@Z@5@@Z).
1>libcrypto.lib(libcrypto-lib-e_capi.obj) : error LNK2019: ссылка на неразрешенный внешний символ __imp__CertOpenStore@20 в функции _capi_open_store.
1>libcrypto.lib(libcrypto-lib-e_capi.obj) : error LNK2019: ссылка на неразрешенный внешний символ __imp__CertCloseStore@8 в функции _capi_find_key.
1>libcrypto.lib(libdefault-lib-winstore_store.obj) : error LNK2001: неразрешенный внешний символ __imp__CertCloseStore@8.
1>libcrypto.lib(libcrypto-lib-e_capi.obj) : error LNK2019: ссылка на неразрешенный внешний символ __imp__CertEnumCertificatesInStore@8 в функции _capi_find_cert.
1>libcrypto.lib(libcrypto-lib-e_capi.obj) : error LNK2019: ссылка на неразрешенный внешний символ __imp__CertFindCertificateInStore@24 в функции _capi_find_cert.
1>libcrypto.lib(libdefault-lib-winstore_store.obj) : error LNK2001: неразрешенный внешний символ __imp__CertFindCertificateInStore@24.
1>libcrypto.lib(libcrypto-lib-e_capi.obj) : error LNK2019: ссылка на неразрешенный внешний символ __imp__CertDuplicateCertificateContext@4 в функции _capi_load_ssl_client_cert.
1>libcrypto.lib(libcrypto-lib-e_capi.obj) : error LNK2019: ссылка на неразрешенный внешний символ __imp__CertFreeCertificateContext@4 в функции _capi_find_key.
1>libcrypto.lib(libdefault-lib-winstore_store.obj) : error LNK2001: неразрешенный внешний символ __imp__CertFreeCertificateContext@4.
1>libcrypto.lib(libcrypto-lib-e_capi.obj) : error LNK2019: ссылка на неразрешенный внешний символ __imp__CertGetCertificateContextProperty@16 в функции _capi_cert_get_fname.
1>libcrypto.lib(libdefault-lib-winstore_store.obj) : error LNK2019: ссылка на неразрешенный внешний символ __imp__CertOpenSystemStoreW@8 в функции _winstore_open.
1>D:\Programming\WebRTC\Debug\WebRTC.exe : fatal error LNK1120: неразрешенных внешних элементов: 33
```
Notes: I'm using vcpkg in global mode, also some of the errors are deleted from Failure log because it was too long

Help would be very appreciated, as I lack understanding of why this happens and am stuck for more than a week

--- a/ports/libdatachannel/library-linkage.diff
+++ b/ports/libdatachannel/library-linkage.diff
@@ -1,8 +1,16 @@
 diff --git a/CMakeLists.txt b/CMakeLists.txt
-index b35e4b1..1b5190b 100644
+index b35e4b1..b052d02 100644
 --- a/CMakeLists.txt
 +++ b/CMakeLists.txt
-@@ -457,11 +457,17 @@ if(WARNINGS_AS_ERRORS)
+@@ -48,7 +48,6 @@ endif()
+ 
+ list(APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake/Modules)
+ set(CMAKE_POSITION_INDEPENDENT_CODE ON)
+-set(BUILD_SHARED_LIBS OFF) # to force usrsctp to be built static
+ 
+ if(WIN32)
+ 	add_definitions(-DWIN32_LEAN_AND_MEAN)
+@@ -457,11 +456,17 @@ if(WARNINGS_AS_ERRORS)
  	endif()
  endif()
  

--- a/ports/libdatachannel/portfile.cmake
+++ b/ports/libdatachannel/portfile.cmake
@@ -33,6 +33,7 @@ vcpkg_fixup_pkgconfig()
 
 if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
     vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/rtc/common.hpp" "#ifdef RTC_STATIC" "#if 1")
+    vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/rtc/rtc.h" "#ifdef RTC_STATIC" "#if 1")
 endif()
 
 file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include" "${CURRENT_PACKAGES_DIR}/debug/share")

--- a/ports/libdatachannel/vcpkg.json
+++ b/ports/libdatachannel/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "libdatachannel",
   "version-semver": "0.21.2",
+  "port-version": 1,
   "description": "libdatachannel is a standalone implementation of WebRTC Data Channels, WebRTC Media Transport, and WebSockets in C++17 with C bindings for POSIX platforms (including GNU/Linux, Android, and Apple macOS) and Microsoft Windows.",
   "homepage": "https://github.com/paullouisageneau/libdatachannel",
   "license": "MPL-2.0",

--- a/scripts/test_ports/vcpkg-ci-libdatachannel/cmake-project.diff
+++ b/scripts/test_ports/vcpkg-ci-libdatachannel/cmake-project.diff
@@ -1,0 +1,29 @@
+diff --git a/examples/streamer/CMakeLists.txt b/examples/streamer/CMakeLists.txt
+index 1bdeb7c..9cf65dd 100644
+--- a/examples/streamer/CMakeLists.txt
++++ b/examples/streamer/CMakeLists.txt
+@@ -2,6 +2,15 @@ cmake_minimum_required(VERSION 3.7)
+ if(POLICY CMP0079)
+ 	cmake_policy(SET CMP0079 NEW)
+ endif()
++project(vcpkg-ci)
++add_definitions(-DWIN32_LEAN_AND_MEAN)
++set(THREADS_PREFER_PTHREAD_FLAG ON)
++find_package(Threads REQUIRED)
++find_package(nlohmann_json CONFIG REQUIRED)
++find_package(LibDataChannel CONFIG REQUIRED)
++if(NOT TARGET LibDataChannel::LibDataChannel)
++	add_library(LibDataChannel::LibDataChannel ALIAS LibDataChannel::LibDataChannelStatic)
++endif()
+ 
+ set(STREAMER_SOURCES
+ 	main.cpp
+@@ -47,7 +56,7 @@ set_target_properties(streamer PROPERTIES
+ find_package(Threads REQUIRED)
+ target_link_libraries(streamer LibDataChannel::LibDataChannel Threads::Threads nlohmann_json::nlohmann_json)
+ 
+-if(MSVC)
++if(0)
+ 	add_custom_command(TARGET streamer POST_BUILD
+ 		COMMAND ${CMAKE_COMMAND} -E copy_if_different
+ 		"$<TARGET_FILE_DIR:datachannel>/datachannel.dll"

--- a/scripts/test_ports/vcpkg-ci-libdatachannel/portfile.cmake
+++ b/scripts/test_ports/vcpkg-ci-libdatachannel/portfile.cmake
@@ -1,0 +1,14 @@
+set(VCPKG_POLICY_EMPTY_PACKAGE enabled)
+
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO paullouisageneau/libdatachannel
+    REF v0.21.2
+    SHA512 fd0d66bb932e29abc01e9f1a8b16ccb79012a7e3901e2e0f882f56ab2f090260945e1556c85ad07ef897b8c70fcdd44cdeead9955a9bca7afe1dda8900c473cc
+    HEAD_REF master
+    PATCHES 
+        cmake-project.diff
+)
+
+vcpkg_cmake_configure(SOURCE_PATH "${SOURCE_PATH}/examples/streamer")
+vcpkg_cmake_build()

--- a/scripts/test_ports/vcpkg-ci-libdatachannel/vcpkg.json
+++ b/scripts/test_ports/vcpkg-ci-libdatachannel/vcpkg.json
@@ -1,0 +1,19 @@
+{
+  "name": "vcpkg-ci-libdatachannel",
+  "version-string": "ci",
+  "description": "Validates libdatachannel",
+  "dependencies": [
+    {
+      "name": "libdatachannel",
+      "features": [
+        "srtp",
+        "ws"
+      ]
+    },
+    "nlohmann-json",
+    {
+      "name": "vcpkg-cmake",
+      "host": true
+    }
+  ]
+}

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -4346,7 +4346,7 @@
     },
     "libdatachannel": {
       "baseline": "0.21.2",
-      "port-version": 0
+      "port-version": 1
     },
     "libdatrie": {
       "baseline": "0.2.13",

--- a/versions/l-/libdatachannel.json
+++ b/versions/l-/libdatachannel.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "df0663ab881ed7a331bf51b6bf652e80fc55dbf4",
+      "version-semver": "0.21.2",
+      "port-version": 1
+    },
+    {
       "git-tree": "7ef5c0cdf31856f7b73f1498dd8055627608801c",
       "version-semver": "0.21.2",
       "port-version": 0

